src/server/builder.ts

```diff
@@ -326,22 +293,24 @@ namespace ts.server {
                 const oldReference = oldReferences[oldIndex];
                 const newReference = newReferences[newIndex];
                 const compare = ModuleBuilderFileInfo.compareFileInfos(oldReference, newReference);
-                if (compare < 0) {
-                    // New reference is greater then current reference. That means
-                    // the current reference doesn't exist anymore after parsing. So delete
-                    // references.
-                    oldReference.removeReferencedBy(fileInfo);
-                    oldIndex++;
-                }
-                else if (compare > 0) {
-                    // A new reference info. Add it.
-                    newReference.addReferencedBy(fileInfo);
-                    newIndex++;
-                }
-                else {
-                    // Equal. Go to next
-                    oldIndex++;
-                    newIndex++;
+                switch (compare) {

```

Comment:
given you have c;leaned this up, is it possible to use enumerate inserted and deleted from utilities instead